examples/tradeable-cashflow-hardhat/contracts/RedirectAll.sol

```diff
@@ -0,0 +1,233 @@
+
+        _host.registerApp(configWord);
+    }
+
+    /**************************************************************************
+     * Redirect Logic
+     *************************************************************************/
+
+    function currentReceiver()
+        external
+        view
+        returns (
+            uint256 startTime,
+            address receiver,
+            int96 flowRate
+        )
+    {
+        if (_receiver != address(0)) {
+            (startTime, flowRate, , ) = _cfa.getFlow(
+                _acceptedToken,
+                address(this),
+                _receiver
+            );
+            receiver = _receiver;
+        }
+    }
+
+    event ReceiverChanged(address receiver); //what is this?
+
+    /// @dev If a new stream is opened, or an existing one is opened
+    function _updateOutflow(bytes calldata ctx)
+        private
+        returns (bytes memory newCtx)
+    {
+        newCtx = ctx;
+        // @dev This will give me the new flowRate, as it is called in after callbacks
+        int96 netFlowRate = _cfa.getNetFlow(_acceptedToken, address(this));
+        (, int96 outFlowRate, , ) = _cfa.getFlow(
+            _acceptedToken,
+            address(this),
+            _receiver
+        ); // CHECK: unclear what happens if flow doesn't exist.
+        int96 inFlowRate = netFlowRate + outFlowRate;
+
+        // @dev If inFlowRate === 0, then delete existing flow.
+        if (inFlowRate == int96(0)) {
+            // @dev if inFlowRate is zero, delete outflow.
+            newCtx = cfaV1.deleteFlowWithCtx(
+                newCtx,
+                address(this),
+                _receiver,
+                _acceptedToken
+            );
+        } else if (outFlowRate != int96(0)) {
+            newCtx = cfaV1.updateFlowWithCtx(
+                newCtx,
+                _receiver,
+                _acceptedToken,
+                inFlowRate
+            );
+        } else {
+            // @dev If there is no existing outflow, then create new flow to equal inflow
+            newCtx = cfaV1.createFlowWithCtx(
+                newCtx,
+                _receiver,
+                _acceptedToken,
+                inFlowRate
+            );
+        }
+    }
+
+    // @dev Change the Receiver of the total flow
+    function _changeReceiver(address newReceiver) internal {
+        require(newReceiver != address(0), "New receiver is zero address");
+        // @dev because our app is registered as final, we can't take downstream apps
+        require(
+            !_host.isApp(ISuperApp(newReceiver)),
+            "New receiver can not be a superApp"
+        );
+        if (newReceiver == _receiver) return;
+        // @dev delete flow to old receiver
+        (, int96 outFlowRate, , ) = _cfa.getFlow(
+            _acceptedToken,
+            address(this),
+            _receiver
+        ); //CHECK: unclear what happens if flow doesn't exist.
```

Comment:
I'm just gonna assume this was checked, looks like a copy pasta from the OG `RedirectAll.sol`

Reply:
Yes copy pasta.